examples/nemo/scripts/omp_gpu_trans.py

```diff
@@ -107,9 +107,11 @@ def trans(psy):
 
         # For performance in lib_fortran, mark serial routines as GPU-enabled
         if psy.name == "psy_lib_fortran_psy":
-            if not invoke.schedule.walk((Loop, Call)):
-                OMPDeclareTargetTrans().apply(invoke.schedule)
-                continue
+            if not invoke.schedule.walk(Loop):
```

Comment:
Tbh, I am not sure I agree with this. Traversals are fast (and repeated traversal presumably have everything in inner level cache) so I would prioritize readability here.

(also if we really needed to make it fast we should use generators so the condition is evaluated for each node on the fly instead of creating temporal lists)